### PR TITLE
ARROW-625: [C++] Add TimeUnit to TimeType::ToString. Add timezone to TimestampType::ToString if present

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

File: cpp/src/arrow/type.cc [hunk 1]
```diff
@@ -108,7 +108,7 @@ std::string Date32Type::ToString() const {
   return std::string("date32");
 }
 
-static inline void print_time_unit(TimeUnit unit, std::ostream* stream) {
+static inline void PrintTimeUnit(TimeUnit unit, std::ostream* stream) {
```
Comment (tebeka) [inline cpp/src/arrow/type.cc]: Why not the usual `std::ostream& operator<<(std::ostream& os, TimeUnit t)` ?

Comment (wesm) [inline cpp/src/arrow/type.cc]: done

**Review by xhochy — APPROVED** [review]
+1, LGTM. Merge on green build.

**Comment by wesm** [comment]
OS X stalled again, hopefully fixed by #380. Merging

**Pull request closed without merging**